display/type340.c

```diff
@@ -711,37 +731,47 @@ character(int n, unsigned char c)
     struct type340 *u = UNIT(0);
     int x, y;
     unsigned char s = u->scale;
+    unsigned char flags;
 
-    switch (c) {
-    case CHRLF:                         /* LF */
-        u->ypos -= 12*s;
+    c |= u->shift;
+    flags = chars[c][5];
+
+    if (flags & CH_LF) {                /* LF */
+        u->ypos -= u->height*s;
         if (u->ypos < 0) {
             u->status |= ST340_HEDGE;
             u->ypos = 0;
         }
         return 0;
-    case CHRCR:                         /* CR */
+    }
+    if (flags & CH_CR) {                /* CR */
         u->xpos = 0;
         return 0;
-    case CHRUC:                         /* "SHIFT IN (HORIZ)" */
+    }
+    if (flags & CH_UC) {                /* "SHIFT IN (HORIZ)" */
         u->shift = 0;                   /* upper case in SPCWAR 163 */
         return 0;
-    case CHRLC:                         /* "SHIFT OUT (VERT)" */
+    }
+    if (flags & CH_LC) {                /* "SHIFT OUT (VERT)" */
         u->shift = 0100;                /* lower case in SPCWAR 163 */
         return 0;
-    case CHRESC:                        /* escape */
+    }
+    if (flags & CH_ESC) {               /* escape */
         return 1;
     }
+    if ((flags & CH_NSPC) && u->xpos >= u->width*s) {

```

Comment:
Seems ugly (to me) to subtract after the fact; wrap the increment in `!(flags &CH_NSP)` instead?

Reply:
It seems the nonspacing characters backspace and overstrike the character before.  This decrement is not after the fact, but before the character is put on the display.  The increment happens after.